packages/alfa-aria/src/node.ts

```diff
@@ -179,48 +231,48 @@ export namespace Node {
     return Branched.of(Inert.of(node));
   }
 
-  function build(
-    node: dom.Node,
-    device: Device,
-    parent: Option<Node> = None
-  ): Branched<Node, Browser> {
+  function build(node: dom.Node, device: Device): Branched<Node, Browser> {
     return cache.get(device, Cache.empty).get(node, () => {
-      let accessibleNode: Branched<Node, Browser>;
-
       // Text nodes are _always_ exposed in the accessibility tree.
       if (dom.Text.isText(node)) {
-        accessibleNode = Branched.of(
-          Text.of(node, node.data.replace(/\s+/g, " "))
-        );
+        return Name.from(node, device).map((name) => Text.of(node, name.get()));
       }
 
+      const children = () =>
+        Branched.traverse(node.children({ flattened: true }), (child) =>
+          build(child, device)
+        );
+
       // Element nodes are _sometimes_ exposed in the accessibility tree.
-      else if (dom.Element.isElement(node)) {
-        // Elements that are explicitly excluded from the accessibility tree by
-        // means are `aria-hidden=true` are never exposed in the accessibility
-        // tree, nor are their descendants.
+      if (dom.Element.isElement(node)) {
+        // Elements that are explicitly excluded from the accessibility tree
+        // by means of `aria-hidden=true` are never exposed in the
+        // accessibility tree, nor are their descendants.
         //
-        // This behaviour is unfortunately not consistent across browsers, which
-        // we may or may not want to deal with. For now, we pretend that all
-        // browsers act consistently.
+        // This behaviour is unfortunately not consistent across browsers,
+        // which we may or may not want to deal with. For now, we pretend that
+        // all browsers act consistently.
         //
         // https://github.com/Siteimprove/alfa/issues/184#issuecomment-593878009
         if (
           node
             .attribute("aria-hidden")
-            .some((attr) => attr.value.toLowerCase() === "true")
+            .some((attribute) =>
+              attribute.enumerate("true", "false").some(equals("true"))
```

Comment:
Make sense.
I feel there is two layers in the roles definition. One is the whole structure of roles: a thingie with sub/super classes, required context, supported attributes, … The other is a particular instance of that structure where we have a bunch of actual roles.
It is possible to define a full new roles hierarchy with actual roles being totally different from ARIA. This other hierarchy would still require `isSuperClass` or `isSupported` but wouldn't have `isLandmark` because maybe it wouldn't have a role called "landmark" in the first place.
That's why I feel maybe `isLandmark` and friends could be move to a separate layer of code: they only make sense for the particular hierarchy of roles that we consider here.

Reply:
> Doesn't TS already know that aria-hidden is a boolean?

Not quite; it's just another DOM attribute with a `string` value.